### PR TITLE
Adding option to hide unlisted 7TV emotes and show them by default

### PR DESCRIPTION
**Pull request by igorovh**

### CODE REVIEW

**Comment by night** [comment]
I am not super familiar with the 7TV platform. What is an `unlisted` emote?

**Comment by igorovh** [comment]
It is not public emote, like [jasperKiss](https://7tv.app/emotes/60b15b3ec5e3fb453e887716). It is not verified by 7TV, but it still can be used on chat. There are a lot of big and small communities that have own emotes based around streamer that are not verified. This is why that options is important for them.

File: src/modules/seventv/channel-emotes.js [hunk 1]
```diff
@@ -89,9 +91,10 @@ class SevenTVChannelEmotes extends AbstractEmotes {
     let message;
     switch (action) {
       case 'ADD': {
-        if (isUnlisted(emote.visibility)) {
-          return;
-        }
+        const hideUnlisted =
+          hasFlag(settings.get(SettingIds.EMOTES), EmoteTypeFlags.SEVENTV_HIDE_UNLISTED) &&
+          isUnlisted(emote.visibility);
```
Comment (night) [inline src/modules/seventv/channel-emotes.js:96]: this seems to filter unlisted and permanently unlisted emotes. what is the difference between unlisted and permanently unlisted?

File: src/constants.js [hunk 2]
```diff
@@ -73,6 +73,7 @@ export const EmoteTypeFlags = {
   ANIMATED_PERSONAL_EMOTES: 1 << 3,
   SEVENTV_EMOTES: 1 << 4,
   EMOTE_MODIFIERS: 1 << 5,
+  SEVENTV_HIDE_UNLISTED: 1 << 6,
```
[the remark this answers is not in the document]
Comment (userMacieG) [reply src/constants.js:76]: In 7tv unlisted are shown by default, They have option to hide them, so doesnt make sense to invert that setting.

File: src/modules/seventv/channel-emotes.js (hunk 1 above)
Comment (userMacieG) [reply src/modules/seventv/channel-emotes.js:96]: No idea, can't find anything about that in their code or on website.

Comment (mmattbtw) [reply src/modules/seventv/channel-emotes.js:96]: iirc: `unlisted` = hasn't been vetted/verified by 7TV moderators
permanently unlisted = vetted by 7TV moderators to not be shown on public listings, can still be added to channels, but can only be accessed via a link.

**Pull request merged**